### PR TITLE
Tried to fix #108

### LINKED ISSUE #108: Chore: Merge remote file adapter to core: change from Phoenix
Merge in the following changes from phoenix for remote file improvements: 
* https://github.com/aicore/phoenix/pull/274
* https://github.com/aicore/phoenix/pull/275

## This brings in
* Move remote file adapter to core from extension as reading HTTP/HTTPS remote content as a file is a core requirement that any extension should be able to use. 
* Added support for file reads on any browser supported encoding from HTTP/HTTPS links instead of just defaulting to utf-8 reads
* Adds support for github and gitlab file URL opening with https://github.com/aicore/phoenix/pull/275

### PR DESCRIPTION
**Pull request by TatoAndTatoOnly**
I might have messed something up really badly- basically, I did Pheonix pulls [#274](https://github.com/aicore/phoenix/pull/274) and [#275](https://github.com/aicore/phoenix/pull/275).

### CODE REVIEW

**Comment by abose** [comment]
Many thanks @TatoAndTatoOnly for contributing to brackets.

MIT re-licencing rights has been granted to this pull request from phoenix: https://github.com/aicore/phoenix/issues/298

File: src/extensions/default/RemoteFileAdapter/main.js [hunk 1]
```diff
@@ -1,46 +1,42 @@
 /*
- * Copyright (c) 2018 - present Adobe Systems Incorporated. All rights reserved.
+ * GNU AGPL-3.0 License
```
Comment (abose) [inline src/extensions/default/RemoteFileAdapter/main.js:2]: Please change license to MIT, with copyright the line 
```
Modified Work Copyright (c) 2021 - present core.ai & brackets.io. All rights reserved.
Original work Copyright (c) 2018 - 2021 Adobe Systems Incorporated. All rights reserved.
```

**Review by abose — CHANGES_REQUESTED** [review]

**Comment by abose** [comment]
In Further pull requests, it is good to have the following sessions:

1. How was this tested? Eg. how to test steps and observations.
2. A gif screen recording to help the UI changes. you can use [peek](https://github.com/phw/peek) in linux or [ScreenToGif](https://github.com/NickeManarin/ScreenToGif/) in windows

**Comment by TatoAndTatoOnly** [comment]
I've changed the copyright and added the MIT license. I'll start testing soon, hopefully.

File: src/extensions/default/RemoteFileAdapter/main.js (hunk 1 above)
Comment (TatoAndTatoOnly) [reply src/extensions/default/RemoteFileAdapter/main.js]: Fixed (hopefully).

Comment (TatoAndTatoOnly) [reply src/extensions/default/RemoteFileAdapter/main.js]: Wait a second- not fixed.

Comment (TatoAndTatoOnly) [reply src/extensions/default/RemoteFileAdapter/main.js]: Wait another second- fixed.

**Review by TatoAndTatoOnly — COMMENTED** [review]
Didn’t mean to do that.

File: src/extensions/default/RemoteFileAdapter/main.js (hunk 1 above)
Comment (TatoAndTatoOnly) [reply src/extensions/default/RemoteFileAdapter/main.js]: No, wasn’t fixed, but is now.

File: src/extensions/default/RemoteFileAdapter/main.js [hunk 2]
```diff
@@ -1,46 +1,34 @@
...
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
- * DEALINGS IN THE SOFTWARE.
- *
- */
+/*Modified Work Copyright (c) 2021 - present core.ai & brackets.io. All rights reserved.
```
Comment (abose) [inline src/extensions/default/RemoteFileAdapter/main.js:1]: The formatting is off in this. The only change that needs to be added here is to add the new copyright string and everything else can be left as in in the license.

File: src/filesystem/FileSystem.js [hunk 3]
```diff
@@ -1,23 +1,21 @@
 /*
- * Copyright (c) 2013 - present Adobe Systems Incorporated. All rights reserved.
+ * GNU AGPL-3.0 License
```
Comment (abose) [inline src/filesystem/FileSystem.js:2]: GPL licence to MIT

**Review by abose — CHANGES_REQUESTED** [review]
Please also squash all the commits to a single one.

**Comment by TatoAndTatoOnly** [comment]
Something's gone wrong and I'm having a lot of trouble squashing the commits, but it should be possible to do that on merge.

File: src/filesystem/FileSystem.js (hunk 3 above)
Comment (TatoAndTatoOnly) [reply src/filesystem/FileSystem.js]: I think these should be done.

**Comment by abose** [comment]
Superseded by https://github.com/brackets-cont/brackets/pull/188

**Comment by abose** [comment]
Thanks @TatoAndTatoOnly for addressing this issue. Closing as the pull request functionality is merged as part of https://github.com/brackets-cont/brackets/pull/188

**Pull request closed without merging**